src/main/scala/chiseltest/package.scala

```diff
@@ -205,7 +205,23 @@ package object chiseltest {
       case x => throw new LiteralTypeException(s"don't know how to peek $x")
     }
 
-    def peek(): T = peekWithStale(false)
+    protected def peekWithStaleBool(stale: Boolean): Boolean = x match {
+      case (x: Bool) =>
+        Context().backend.peekBits(x, stale) match {
+          case x: BigInt if x == 0 => false
+          case x: BigInt if x == 1 => true
+          case x => throw new LiteralTypeException(s"peeked Bool with value $x not 0 or 1")
+        }
+    }
+
+    protected def peekWithStaleBigInt(stale: Boolean): BigInt = x match {
+      case (x: UInt) => Context().backend.peekBits(x, stale)
+      case (x: SInt) => Context().backend.peekBits(x, stale)
+    }
+
+    def peek():        T = peekWithStale(false)
+    def peekBoolean(): Boolean = peekWithStaleBool(false)
+    def peekInt():     BigInt = peekWithStaleBigInt(false)
```

Comment:
It would be best if these weren't available on `Aggregates`, could these be extension methods on `T <: Element` instead of the current `T <: Data`?

Reply:
Humm.. might need some guidance here :)